types/types.hh

```diff
@@ -1002,3 +1002,52 @@ struct appending_hash<data_type> {
         feed_hash(h, v->name());
     }
 };
+
+class data_values_map {
+    std::unordered_map<std::string, data_value> _map;
+
```

Comment:
Why std::string and not sstring?

Why a wrapper over a map? Can use the map type directly.

Reply:
Why sstring?  I thought we're trying to stick with the standard when there's no special reason to use sstring (like the uninitialized construct case)

Reply:
> Why a wrapper over a map? Can use the map type directly.

To make the update case simpler.
With bare map `m["foo"] = "bar"` doesn't work because data_value can't be constructed this way.

Reply:
unordered_map::insert_or_assign()

If you're solving a common problem, look for a common solution.

Reply:
> Why sstring? I thought we're trying to stick with the standard when there's no special reason to use sstring (like the uninitialized construct case)

We're using sstring over std::string to avoid conversions when we do need sstring.

Reply:
I thought the main argument for `sstring` over `std::string` is performance, since the former doesn't have to be thread safe. Isn't this the case?

Reply:
Recent (C++11) std::string is also not using copy-on-write. But it's still slower than sstring (sstring doesn't have a separate capacity word).

So we use sstring where performance is important, but also generally to avoid conversions.